src/legacy/ui/public/filter_bar/query_filter.js

```diff
@@ -28,9 +28,19 @@ import { mapAndFlattenFilters } from './lib/map_and_flatten_filters';
 import { extractTimeFilter } from './lib/extract_time_filter';
 import { changeTimeFilter } from './lib/change_time_filter';
 
-export function FilterBarQueryFilterProvider(Promise, indexPatterns, $rootScope, getAppState, globalState, config) {
+import { getNewPlatform } from 'ui/new_platform';
+
+export function FilterBarQueryFilterProvider(Promise, indexPatterns, $rootScope, getAppState, globalState) {
   const queryFilter = {};
 
+  // ui settings handling
+  const uiSettingsClient = getNewPlatform().setup.core.uiSettings;
+  const pinnedByDefault$ = uiSettingsClient.get$('filters:pinnedByDefault');
+  let pinnedByDefault;
+  pinnedByDefault$.subscribe((newValue) => {
+    pinnedByDefault = newValue;
```

Comment:
Just curious about how this async code works. Seems odd to subscribe to an observable and use the resulting value in another method. I'm sure I'm missing something.

Reply:
I'm not an rxjs whiz by any means, but aren't scenarios like this what `toPromise` is often used for?

Since we are just setting a single value, this feels more readable (assuming it does what I think it does)
```javascript
const pinnedByDefault = await pinnedByDefault$.first().toPromise();
```

Reply:
Went with @streamich suggestion